OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1645,16 +1669,9 @@ int Model::replaceMarkerSet(const SimTK::State& s, MarkerSet& aMarkerSet)
     // Now add the markers from aMarkerSet whose body names match bodies in the engine.
     for (i = 0; i < aMarkerSet.getSize(); i++)
     {
-        // Eran: we make a *copy* since both _markerSet and aMarkerSet own their elements (so they will delete them)
-        Marker* marker = aMarkerSet.get(i).clone();
-        const string& frameName = marker->getFrameName();
-        if (getFrameSet().contains(frameName))
-        {
-            const OpenSim::PhysicalFrame* frame = dynamic_cast<const PhysicalFrame*>(&getFrameSet().get(frameName));
-            if(frame) marker->changeFrame(*frame);
-            upd_MarkerSet().adoptAndAppend(marker);
-            numAdded++;
-        }
+        Marker* marker = aMarkerSet[i].clone();
+        upd_MarkerSet().adoptAndAppend(marker);
+        ++numAdded;
```

Comment:
It is a good point, but I believe the previous implementation was not working as expected. Silently not including a `Marker` because its frame name was not in the `FrameSet` (which few are, most Markers are associated with Body frames not Frames in the Model's FrameSet property!) or was misspelled would lead to strange behavior without warning/notification to the user. Now we treat Marker as the Component it is and rely on the Socket mechanism to handle the parent frame dependency uniformly.